components/coins-status/index.js

```diff
@@ -1,43 +1,88 @@
-import { useEffect } from 'react';
+import { useState, useEffect } from 'react';
 import { useAnimateValue } from './useCoinStatus';
 import styles from './coin.module.css';
 import PropTypes from 'prop-types';
 
+const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
+const WALLET_URL = `${BASE_API_URL}/wallet`;
+const CURRENCIES = {
+  NEELAM: 'neelam',
+  DINERO: 'dinero',
+};
+
 const Coins = (props) => {
-  const [coins, changeCoins] = useAnimateValue(0, props.coin.value, 5000);
+  const { coin, balance } = props;
+  const [coins, changeCoins] = useAnimateValue(0, balance, 5000);
+  const currencyColor =
+    coin === CURRENCIES.NEELAM
+      ? styles.blue
+      : coin === CURRENCIES.DINERO
+      ? styles.green
+      : styles.gray;
+
   useEffect(() => {
-    changeCoins(props.coin.value);
-  }, [props.coin.value, changeCoins]);
+    changeCoins(balance);
+  }, [balance, changeCoins]);
+
   return (
     <div className={styles.cointypeIndicator}>
       <div className={styles.coinData}>
-        <p> {props.coin.name}</p>
-        <p> {coins} </p>
+        <p>{coin}</p>
+        <p>{balance}</p>
       </div>
       <div className={`coin ${styles.rotateVertCenter}`}></div>
       <style jsx>{`
         .coin {
-          background-color: ${props.coin.color};
-          width: calc(4.25em / 2);
-          height: calc(4.25em / 2);
+          background-color: ${currencyColor};
+          width: calc(5em / 2);
+          height: calc(5em / 2);
           border-radius: 50%;
           box-shadow: 3px 3px 3px #000;
-          border: 1px solid ${props.coin.borderColor};
+          border: 1px solid ${currencyColor};
         }
       `}</style>
     </div>
   );
 };
 
-const CoinsStatus = (props) => {
-  let coinsArray = props.coins.map((coin, index) => {
-    return <Coins coin={coin} key={index} />;
-  });
-  return <div className={styles.coinsContainer}>{coinsArray}</div>;
+const CoinsStatus = () => {
+  const [currencies, setCurrencies] = useState([]);
+  const coins = [];
+
+  useEffect(async () => {
```

Comment:
Effect callbacks are synchronous to prevent race conditions. Let's move the async fetch function outside.

```
async function fetchData() {
  const response = await MyAPI.getData(someId);
  // ...
}

useEffect(() => {
  fetchData();
}, []);
```

More here: [Data fetching with Hooks](https://reactjs.org/link/hooks-data-fetching)

Eslint error: (react-hooks/exhaustive-deps)

Reply:
Even after moving the async function outside of useEffect hook, still Eslint showing same warning message.

Reply:
That's because of a separate reason that can be solved using useCallback

Reply:
Here's why: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies